### PR TITLE
upgrade to luigi 2.3.2

### PR DESCRIPTION
**Pull request by kaloramik**
- Updates luigi to 2.3.2 
- adds an index on task_id in task_events (copied from luigi)
- adds an index on event_name in task_events (There are queries that could use this, want to see if this makes things faster)
- add task_id in tasks (copied from luigi, although luigi has String(200), and our task_ids overflows that)

### CODE REVIEW

File: fireflower/profiler.py [hunk 1]
```diff
@@ -41,7 +41,11 @@ def profiling(logger, key=None, scale=1, to_profile=True, **kwargs):
         log_kwargs['max_mem_used'] = humanize.naturalsize(after_max_mem_usage - before_max_mem_usage)
         level = logging.INFO if num_secs > 0.1 else logging.DEBUG
 
-        logger.log(level, "profiling_end", **log_kwargs)
+        if level == logging.INFO:
+            logger.info("profiling_end", **log_kwargs)
+        elif level == logging.DEBUG:
+            logger.debug("profiling_end", **log_kwargs)
+
```
Comment (brianmartin) [inline fireflower/profiler.py]: hm, wasn't working when level passed into `logger.log`?

File: fireflower/models.py [hunk 2]
```diff
@@ -57,8 +57,8 @@ class TaskEvent(FireflowerDeclBase):
     __tablename__ = 'task_events'
 
     id = Column(Integer, primary_key=True)
-    task_id = Column(Integer, ForeignKey('tasks.id'))
-    event_name = Column(String(20))
+    task_id = Column(Integer, ForeignKey('tasks.id'), index=True)
+    event_name = Column(String(20), index=True)
```
Comment (brianmartin) [inline fireflower/models.py]: add migrations

File: requirements.txt [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+luigi==2.3.2
+SQLAlchemy==1.1.1
 toolz==0.7.2
 pandas==0.17.1
 nose==1.3.7
 nose-parameterized==0.5.0
 structlog==16.0.0
 testfixtures==4.1.2
-moto==0.4.2
+moto==0.4.25
```
Comment (brianmartin) [inline requirements.txt]: will need docker build

File: fireflower/profiler.py (hunk 1 above)
Comment (kaloramik) [inline fireflower/profiler.py]: it throws an error, I think its because logger.log doesnt actually work by default? it works if u use structlog

**Comment by brianmartin** [comment]
Oops, didn't even realize I was in the fireflower repo instead of signals. #goodreviewer

**Comment by birbbit** [comment]
We can use luigi.ListParameter instead of ListParameter now!

**Pull request merged**